src/Eshopworld.WorkerProcess/WorkerLease.cs

```diff
@@ -80,45 +73,28 @@ public void StopLeasing()
 
         internal async Task LeaseAsync()
         {
-            _timer.Stop();
-
-            try
+            if (CheckLeaseExpired(CurrentLease))
             {
-                if (CheckLeaseExpired(CurrentLease))
-                {
-                    await _leaseAllocator.ReleaseLeaseAsync(CurrentLease).ConfigureAwait(false);
-
-                    OnLeaseExpired();
-                }
-
-                CurrentLease = await _leaseAllocator.AllocateLeaseAsync(InstanceId).ConfigureAwait(false);
-
-                if (CurrentLease != null)
-                    OnLeaseAllocated(CurrentLease.LeasedUntil.GetValueOrDefault());
-
-                if(CurrentLease?.LeasedUntil.HasValue ?? false)
-                {
-                    _timer.Interval = CurrentLease.Interval.Value.TotalMilliseconds;
-                }
-                else
-                {
-                    _timer.Interval = _slottedInterval.Calculate(
-                        CurrentLease?.LeasedUntil ?? ServerDateTime.UtcNow,
-                        _options.Value.LeaseInterval).TotalMilliseconds;
-                }
+                await _leaseAllocator.ReleaseLeaseAsync(CurrentLease).ConfigureAwait(false);
```

Comment:
Did we say this method has to be removed? Or is that covered by another PR (cc: @nikhilgirraj )

Reply:
that was my understanding. Is that correct @nikhilgirraj ?